app/services/search/base.rb

```diff
@@ -43,6 +43,19 @@ def search(body:)
         Search::Client.search(index: self::INDEX_ALIAS, body: body)
       end
 
+      def set_query_size(params)
```

Comment:
I'm not a big fan of methods that modify an object like this, they can be harder to re-use from other places and it's hard to know what `params` are needed without looking at the method body. I prefer something like:

```ruby
def build_search_params(page: self::DEFAULT_PAGE, per_page: self::DEFAULT_PER_PAGE)
  {
    page: page,
    per_page: per_page,
    size: per_page.to_i * page.to_i
  }
end
```

Then the caller can do something like 

```ruby
search_params = build_search_params(**params)
```

Anyway, it's mostly a personal preference, but the general pattern has served me well in the past.